smart-contracts/contracts/BridgeBank/BridgeBank.sol

```diff
@@ -16,27 +17,30 @@ import "./WhiteList.sol";
  *      list that can be locked.
  **/
 
-contract BridgeBank is CosmosBank, EthereumBank, WhiteList {
-    using SafeMath for uint256;
+contract BridgeBank is BankStorage, CosmosBank, EthereumBank, WhiteList {
 
-    address public operator;
-    Oracle public oracle;
-    CosmosBridge public cosmosBridge;
-    address public owner;
+    bool private _initialized;
+
+    using SafeMath for uint256;
 
     /*
-     * @dev: Constructor, sets operator
+     * @dev: Initializer, sets operator
      */
-    constructor(
+    function initialize(
```

Comment:
not sure about versioning (truffle/openzepplin/solc) but it was available inheritance from "initializable" which implements this initialise functionality and adds specific to language syntax "public initialisable" in the end of constructor. Whether to use it or not - I don't have strong opinion but you could check by yourself and make sure how it works

Reply:
I would have used the openzeppelin implementation of initializable, however, it was implemented as an abstract contract which gave a compiler error with the current version of solc that we are using.

Reply:
I believe compilation error is normal behaviour (public initialisable after constructor where "initialisable" is not real modifier). It should be compiled by openzeppelin cli. It is kind of "hack" in solc language. At least how I remember it. There is package @openzeppelin/cli which can do this in scripting as well. But I think your approach however better because it provides more direct control and flexibility.

Reply:
I would use openzeppelin cli, but the problem with that is this project is done in truffle so I don't feel like that would make sense. Additionally, truffle will recompile if the build folder is built by another compiler that doesn't put out the artifacts it's expecting.